homeassistant/components/apcupsd/__init__.py

```diff
@@ -48,7 +48,8 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    hass.data[DOMAIN].pop(entry.entry_id)
+    if unload_ok and DOMAIN in hass.data:
```

Comment:
The DOMAIN will always be in `hass.data` here in production since the config entry must be setup to be unloaded.

We should not add production code to just handle a case that can happen in tests. This is a sign that test code should be fixed instead. Eg, by patching `async_unload_entry` when patching `async_setup_entry`.